backend/src/user.ts

```diff
@@ -29,6 +29,18 @@ export class User {
 			return indices;
 		}, new Array<number>());
 	}
+	getIncorrect(): number[] {
+		return this.answers.reduce((indices, ans, i) => {
+			if (!ans.correct) indices.push(i);
+			return indices;
+		}, new Array<number>());
+	}
+	getMissing(): number[] {
+		return this.answers.reduce((indices, ans, i) => {
+			if (typeof ans === 'undefined') indices.push(i);
```

Comment:
`this.answers` is an array of `AnswerObj` not an array of `AnswerObj|undefined` so it shouldn't ever contain undefined values unless some other code ignored enough type errors to put `undefined` in there.